### ISSUE TITLE
s3 getObject `SignatureDoesNotMatch` if path contains a `:`

### ISSUE THREAD

**Issue by VerTiGoEtrex** [agilord/aws_client#402]
Easy repro — upload a file named `test:file.json` with any contents and attempt to fetch it using s3.getObject.

Request will fail with `SignatureDoesNotMatch`.

I'm going to look into this briefly to see if I can find out why `:` breaks signing, but don't claim to be an AWS expert. Any experienced S3 people have any pointers/ideas on this one?

**Comment by btanev** [comment]
I got the same error `SignatureDoesNotMatch` with S3 putObject.
I've debugged the issue and got the same conclusion as @VerTiGoEtrex as the cause of the problem "uri.path is encoded twice in the canonical hash".

Adding the comment here just to bump the issue as it's an important one. Without the fix (there is already a PR https://github.com/agilord/aws_client/pull/403 ) we cannot use this library to get or put S3 objects in characters != that the standard latin chars and common digits.
